_ingest-pipelines/processors/fail.md

```diff
@@ -0,0 +1,154 @@
+
+The following is the syntax for the `fail` processor:
+
+```json
+"fail": { 
+  "if": "ctx.foo == 'bar'", 
+  "message": "Custom error message" 
+  }
+```
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `fail` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+`message` | Required | Custom error message to be included in the failure response.
+`description`  | Optional  | A brief description of the processor.  |  
+`if` | Optional | A condition for running the processor. |  
+`ignore_failure` | Optional | Specifies whether the processor continues execution even if it encounters errors. If set to `true`, failures are ignored. Default is `false`. |  
+`on_failure` | Optional | A list of processors to run if the processor fails. |  
+`tag` | Optional | An identifier tag for the processor. Useful for debugging in order to distinguish between processors of the same type. |  
+
+## Using the processor
+
+Follow these steps to use the processor in a pipeline.
+
+### Step 1: Create a pipeline
+
+The following query creates a pipeline, named `fail-log-pipeline`, that uses the `fail` processor to intentionally fail the pipeline execution for log events: 
+
+```json
+PUT _ingest/pipeline/fail-log-pipeline  
+{  
+  "description": "A pipeline to test the fail processor for log events",  
+  "processors": [  
+    {  
+      "fail": {  
+        "if": "ctx.user_info.contains('password') || ctx.user_info.contains('credit card')",  
+        "message": "Document containing personally identifiable information (PII) cannot be indexed!"  
+      }  
+    }  
+  ]  
+}
+```
+{% include copy-curl.html %}
+
+### Step 2 (Optional): Test the pipeline
+
+It is recommended that you test your pipeline before you ingest documents.
+{: .tip}
+
+To test the pipeline, run the following query:
+
+```json
+POST _ingest/pipeline/fail-log-pipeline/_simulate  
+{  
+  "docs": [  
+    {  
+      "_source": {  
+        "user_info": "Sensitive information including credit card"  
+      }  
+    }  
+  ]  
+}  
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The following example response confirms that the pipeline is working as expected:
+
+```json
+{
+  "docs": [
+    {
+      "error": {
+        "root_cause": [
+          {
+            "type": "fail_processor_exception",
+            "reason": "Document containing personally identifiable information (PII) cannot be indexed!"
+          }
+        ],
+        "type": "fail_processor_exception",
+        "reason": "Document containing personally identifiable information (PII) cannot be indexed!"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Step 3: Ingest a document 
+
+The following query ingests a document into an index named `logstash-logs`:
```

Comment:
Change the index name to `testindex1` to make it consistent with the name in the query.